src/lib.rs

```diff
@@ -20,21 +20,33 @@
 #![crate_name = "gdal"]
 #![crate_type = "lib"]
 
-pub use version::version_info;
-
-pub mod config;
 pub mod errors;
-mod gdal_major_object;
-pub mod metadata;
-pub mod raster;
-pub mod spatial_ref;
-mod utils;
-pub mod vector;
-pub mod version;
+pub mod utils;
+mod gdal_common;
+pub use gdal_common::*;
+
+#[cfg(any(feature = "gdal_2_2", feature = "gdal_2_3", feature = "gdal_2_4"))]
+pub mod gdal_2_2;
+#[cfg(any(feature = "gdal_2_2", feature = "gdal_2_3", feature = "gdal_2_4"))]
+pub use gdal_2_2::*;
+
+#[cfg(any(feature = "gdal_2_3", feature = "gdal_2_4"))]
+pub mod gdal_2_3;
+#[cfg(any(feature = "gdal_2_3", feature = "gdal_2_4"))]
+pub use gdal_2_3::*;
```

Comment:
I am confused about this part. So why do you import gdal 2.3 stuff for gdal 2.4?
I though all common implementation is done in *common*.

Reply:
There is functionality which is exclusive to the 2.x or 3.x branches, so its not common. However, i'm not aware of functionality in gdal 2.x not available in gdal 2.y where y > x.